### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jonathonpday**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+
+const resetButton = document.querySelector('button[data-behaviour="reset"]');
+const sizeButton = document.querySelector('button[data-behaviour="size"]');
+const rainbowButton = document.querySelector('button[data-behaviour="rainbow"]');
+const blackButton = document.querySelector('button[data-behaviour="black"]');
+const gameContainer = document.querySelector('.gameContainer');
```
Comment (krisquigley) [inline script.js]: Use a data attributes here

File: script.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+//Create Grid
+function grid(size) {
+  for(let i = 1; i <= (size * size); i++) {
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('grid');
+    gameContainer.appendChild(gridCell);
+    document.querySelectorAll('grid');
```
Comment (krisquigley) [inline script.js]: What does this line do

File: script.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  isRainbowPen = false
+});
+
+
+
+//Create Grid
+function grid(size) {
+  for(let i = 1; i <= (size * size); i++) {
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('grid');
```
Comment (krisquigley) [inline script.js]: I would choose a more descriptive css class name

File: script.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
+
+const resetButton = document.querySelector('button[data-behaviour="reset"]');
```
Comment (krisquigley) [inline script.js]: Behaviour should use the American spelling

File: script.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    document.querySelectorAll('grid');
+    gridCell.addEventListener('mouseover', () => colorGridCell(gridCell));
+    gridCell.addEventListener('click', () => gridCell.classList.remove('paint'));
+  }
+}
+grid(16);
+
+//Resets the game and prompts the user for a numerical input for the size of the new grid.
+function makeGrid() {
+  let gridSize = prompt('How many squares would you like on each axis of the new grid?', 16, )
```
Comment (krisquigley) [inline script.js]: Should be a const

File: script.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  document.body.style.setProperty('--gridSize', gridSize)
+  let elementRemove = document.getElementsByClassName('grid')
+  while (elementRemove[0]) {
```
Comment (krisquigley) [inline script.js]: Can you find a more performant way of doing this? Loops will always be more expensive and only used when necessary

File: script.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+//Create Grid
+function grid(size) {
+  for(let i = 1; i <= (size * size); i++) {
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('grid');
+    gameContainer.appendChild(gridCell);
+    document.querySelectorAll('grid');
+    gridCell.addEventListener('mouseover', () => colorGridCell(gridCell));
+    gridCell.addEventListener('click', () => gridCell.classList.remove('paint'));
```
Comment (krisquigley) [inline script.js]: Is this still needed?

File: script.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    penColor = (`rgb(${r}, ${g}, ${b})`);
+  }
+  return penColor
+}
+
+function colorGridCell(gridCell) {
+  gridCell.style.setProperty('background-color', penColor());
+}
+
+resetButton.onclick = reloadPage;
```
Comment (krisquigley) [inline script.js]: Does this work?

File: sketch.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  </head>
+  <body>
+    <h1>Etch-A-Sketch</h1>
+
+    <button class='reset' data-behaviour='reset'>Reset</button>
+    <button class='size' data-behaviour='size'>Size</button>
+    <button class='rainbow' data-behaviour='rainbow'>Rainbow</button>
+    <button class='black' data-behaviour='black'>Black</button>
+
+    <div class='gameContainer' id='gameContainer'></div>
```
Comment (krisquigley) [inline sketch.html]: What's the I'd used for?

File: style.css [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+  font-weight: 400;
+  src: url('../fonts/trade-winds-v6-latin-regular.eot'); /* IE9 Compat Modes */
+  src: local('Trade Winds'), local('TradeWinds'),
+       url('../fonts/trade-winds-v6-latin-regular.woff2') format('woff2'), /* Super Modern Browsers */
+       url('../fonts/trade-winds-v6-latin-regular.woff') format('woff'), /* Modern Browsers */
+}
+
+:root {
+  --gridSize: 16;
+  --paint: black;
```
Comment (krisquigley) [inline style.css]: Is this needed?

File: style.css [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+.gameContainer {
```
Comment (krisquigley) [inline style.css]: CSS class names should be snake case.  Refactor the CSS to use BEM now

File: script.js (hunk 2 above)
Comment (jonathonpday) [reply script.js]: nothing, agreed - deleted.

File: script.js [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  for(let i = 1; i <= (size * size); i++) {
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('gridDivs');
+    gameContainer.appendChild(gridCell);
+    gridCell.addEventListener('mouseover', () => colorGridCell(gridCell));
+  }
+}
+
+//Resets the game and prompts the user for a numerical input for the size of the new grid.
+function makeGrid(gridSize) {
```
Comment (krisquigley) [inline script.js]: This accepts an argument but is never passed one. Also this function doesn't seem to make a grid, the name is misleading

File: script.js [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+  gridCell.style.setProperty('background-color', penColor());
+}
+
+//Reset game to 16 * 16 grid to start again.
+resetButton.onclick = reloadPage;
+function reloadPage () {
+  window.location.reload();
+}
+
+
```
Comment (krisquigley) [inline script.js]: A lot of lines at the end of this file

File: style.css [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+  src: local('Trade Winds'), local('TradeWinds'),
+       url('../fonts/trade-winds-v6-latin-regular.woff2') format('woff2'), /* Super Modern Browsers */
+       url('../fonts/trade-winds-v6-latin-regular.woff') format('woff'), /* Modern Browsers */
+}
+
+:root {
+  --gridSize: 16;
+}
+
+.gameContainer {
```
Comment (krisquigley) [inline style.css]: Not snake case

**Review by krisquigley — COMMENTED** [review]
Just a few more things to clean up

File: script.js [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('grid_divs');
+    gameContainer.appendChild(gridCell);
+    gridCell.addEventListener('mouseover', () => colorGridCell(gridCell));
+  }
+}
+
+//Prompts the user for a numerical input for the size of the new grid which is created.
+function changeSize() {
+  let gridSize = prompt('How many squares would you like on each axis of the new grid?', 16, )
```
Comment (krisquigley) [inline script.js]: This should be a const

**Review by krisquigley — APPROVED** [review]
Remove the extra lines at the end of the file and change the let to a constant and should be good to go ahead and merge

**Pull request merged**